pkg/util/cloudproviders/azure/azure.go

```diff
@@ -162,6 +205,8 @@ func getHostnameWithConfig(ctx context.Context, config config.Config) (string, e
 		name = fmt.Sprintf("%s.%s", metadata.Name, metadata.ResourceGroupName)
 	case "full":
 		name = fmt.Sprintf("%s.%s.%s", metadata.Name, metadata.ResourceGroupName, metadata.SubscriptionID)
+	case "os_computer_name":
+		name = strings.ToLower(metadata.OsProfile.ComputerName)
 	default:
 		return "", fmt.Errorf("invalid azure_hostname_style value: %s", style)
 	}
```

Comment:
Should this switch case and the style checking be part of a function of the new `metadata` struct? 

```go
metadata.GetFromStyle(style)
```